crates/storage/provider/src/providers/database/mod.rs

```diff
@@ -33,29 +38,50 @@ pub struct ProviderFactory<DB> {
     db: DB,
     /// Chain spec
     chain_spec: Arc<ChainSpec>,
+    /// Snapshot Provider
+    snapshot_provider: Option<Arc<SnapshotProvider>>,
```

Comment:
I think this is a reasonable location for the snapshot integration, otherwise we'd need to integrate it into the BlockchainProvider, which I guess would also be suitable.